opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1440,6 +1451,11 @@ namespace Opm {
                     if (group.hasWell(well_name)) {
                         well->setDynamicThpLimit(well_group_thp);
                     }
+                    const auto& ws = this->wellState().well(well->indexOfWell());
+                    const bool thp_is_limit = ws.production_cmode == Well::ProducerCMode::THP;
```

Comment:
should we check this is a producer first in case we have garbage data in `production_cmode` for a injector? not totally sure whether that is possible though.